Shopify/cli
# Add a sensible default for theme app extension template

### WHY are these changes introduced?

When you run `shopify generate extension --type=theme_app_extension` we generate an extension which is basically empty AND not deployable (the user has to manually remove all `.gitkeep` files).

Also fixes https://github.com/Shopify/internal-cli-foundations/issues/493

### WHAT is this pull request doing?

- Provide a sensible template for theme app extensions with:
  - 1 block
  - 1 snippet
  - 1 locale
  - 1 asset
- it also uses metafields for fun
- now the generated theme app extension can be deployed 

### How to test your changes?

- `bin/create-test-app.js -e theme`
- press `p` and install the app in your dev store
- wait for theme app extension to show you three links
- click on the first one and click "Enable"
- click on the theme editor and add a new section with your theme app extension
- open http://localhost:9292 to see the preview
- if you make a change in the source you should see it reflected

### Post-release steps

- change https://shopify.dev/docs/apps/online-store/theme-app-extensions/getting-started

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
